### ISSUE TITLE
out of memory after previewing 3 pathways

### ISSUE THREAD

**Issue by ngehlenborg** [Caleydo/caleydo#1472]
Happened with the KIRC use case data set.

``` java
!ENTRY org.caleydo.core.view.ViewManager@55a43583 1 0 2013-07-14 11:31:17.985
!MESSAGE Registering view: org.caleydo.view.pathway.GLPathway@308d

!ENTRY org.caleydo.core.view.ViewManager@55a43583 1 0 2013-07-14 11:31:18.846
!MESSAGE Registering view: org.caleydo.view.pathway.GLPathway@30f1

!ENTRY org.caleydo.core.view.ViewManager@55a43583 1 0 2013-07-14 11:31:20.529
!MESSAGE Registering view: org.caleydo.view.pathway.GLPathway@3155
Exception in thread "main-null-Timer0" java.lang.OutOfMemoryError: Java heap space
    at java.awt.image.DataBufferInt.<init>(DataBufferInt.java:75)
    at java.awt.image.Raster.createPackedRaster(Raster.java:470)
    at java.awt.image.DirectColorModel.createCompatibleWritableRaster(DirectColorModel.java:1032)
    at java.awt.image.BufferedImage.<init>(BufferedImage.java:357)
    at com.jogamp.opengl.util.awt.TextureRenderer.init(TextureRenderer.java:628)
    at com.jogamp.opengl.util.awt.TextureRenderer.<init>(TextureRenderer.java:136)
    at com.jogamp.opengl.util.awt.TextureRenderer.<init>(TextureRenderer.java:127)
    at com.jogamp.opengl.util.awt.TextureRenderer.<init>(TextureRenderer.java:112)
    at setvis.BubbleSetGLRenderer.init(BubbleSetGLRenderer.java:50)
    at org.caleydo.view.pathway.GLPathway.init(GLPathway.java:375)
    at org.caleydo.view.pathway.GLPathway.initRemote(GLPathway.java:367)
    at org.caleydo.core.view.opengl.layout.util.multiform.MultiFormRenderer$ViewInfo.init(MultiFormRenderer.java:210)
    at org.caleydo.core.view.opengl.layout.util.multiform.MultiFormRenderer$ViewInfo.setActive(MultiFormRenderer.java:202)
    at org.caleydo.core.view.opengl.layout.util.multiform.MultiFormRenderer.ensureValidRenderer(MultiFormRenderer.java:482)
    at org.caleydo.core.view.opengl.layout.util.multiform.MultiFormRenderer.setElementLayout(MultiFormRenderer.java:715)
    at org.caleydo.core.view.opengl.layout.ElementLayout.setRenderer(ElementLayout.java:450)
    at org.caleydo.view.stratomex.brick.layout.DefaultBrickLayoutTemplate.setStaticLayouts(DefaultBrickLayoutTemplate.java:164)
    at org.caleydo.core.view.opengl.layout.LayoutManager.setStaticLayoutConfiguration(LayoutManager.java:89)
    at org.caleydo.view.stratomex.brick.GLBrick.setBrickLayoutTemplate(GLBrick.java:702)
    at org.caleydo.view.stratomex.brick.GLBrick.init(GLBrick.java:317)
    at org.caleydo.view.stratomex.brick.GLBrick.initRemote(GLBrick.java:407)
    at org.caleydo.view.stratomex.column.BrickColumn.display(BrickColumn.java:684)
    at org.caleydo.view.stratomex.GLStratomex.display(GLStratomex.java:627)
    at org.caleydo.view.stratomex.GLStratomex.displayLocal(GLStratomex.java:502)
    at org.caleydo.core.view.opengl.canvas.AGLView.display(AGLView.java:371)
    at jogamp.opengl.GLDrawableHelper.displayImpl(GLDrawableHelper.java:583)
    at jogamp.opengl.GLDrawableHelper.display(GLDrawableHelper.java:567)
    at com.jogamp.opengl.swt.GLCanvas$2.run(GLCanvas.java:149)
    at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1029)
    at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:904)
    at com.jogamp.opengl.swt.GLCanvas$3.run(GLCanvas.java:161)
    at com.jogamp.opengl.swt.GLCanvas.runInGLThread(GLCanvas.java:855)
select next
```

![screen shot 2013-07-14 at 11 35 41 am](https://f.cloud.github.com/assets/1216518/794430/10c79da0-ec9b-11e2-8334-007be4910b35.png)

**Comment by ngehlenborg** [comment]
The same happens with the GBM demo data set after the 6th pathway (there were only 4 groups, while in KIRC there were 5 and I already had one pathway loaded).

![screen shot 2013-07-14 at 11 42 02 am](https://f.cloud.github.com/assets/1216518/794437/17161e9c-ec9c-11e2-96d4-9d1ba2884efb.png)

**Comment by mstreit** [comment]
won't fix. seems to be the low heap space again.

**Comment by mstreit** [comment]
Bubble sets are the problem.

```
org.caleydo.view.pathway.GLPathway @ 0x7d7c61340          |          344 |    36,162,960 |      2.26%
- org.caleydo.view.pathway.PathwayBubbleSet @ 0x7d7c61b50|           24 |    16,111,256 |      1.01%
- org.caleydo.view.pathway.PathwayBubbleSet @ 0x7d7c62180|           24 |    16,111,256 |      1.01%
- org.caleydo.view.pathway.PathwayBubbleSet @ 0x7d7c627b0|           24 |     3,936,496 |      0.25%
```

@dkalkofen any ideas?

**Comment by ngehlenborg** [comment]
Fixed by @dkalkofen in c9c4b4451cbcceeccb6e32b0027ee15d42b0cfeb.
